include/xgboost/data.h

```diff
@@ -40,7 +40,7 @@ enum class DataType : uint8_t {
 
 enum class FeatureType : uint8_t { kNumerical = 0, kCategorical = 1 };
 
-enum class DataSplitMode : int { kRow = 0, kCol = 1 };
+enum class DataSplitMode : int { kRow = 0, kCol = 1, kColS = 2 };
```

Comment:
I'm not sure this is really a data split mode. Would we ever need encryption for horizontal split?

Reply:
for now, horizontal split does not need encryption. but potentially if HE supports further computation, then yes we could also add encryption to horizontal. This datasplit mode leads to below line 189-193 
" bool IsColumnSplit() const { return (data_split_mode == DataSplitMode::kCol) || (data_split_mode == DataSplitMode::kColS); }

  /** @brief Whether the data is split column-wise with secure computation. */
  bool IsSecure() const { return data_split_mode == DataSplitMode::kColS; }
" 
Maybe in the future we can call it "data mode" instead of "data split mode"?